src/pages/[gatewayUID]/node/[nodeId]/details.tsx

```diff
@@ -33,33 +37,198 @@ interface SparrowQueryInterface extends ParsedUrlQuery {
   minutesBeforeNow?: string; // this value is a string to it can be a query param
 }
 
-type NodeDetailsData = {
-  viewModel: NodeDetailViewModel;
-  err?: string;
-};
+const NodeDetails: NextPage = () => {
+  const [viewModel, setViewModel] = useState<NodeDetailViewModel>({});
+  const [isLoading, setIsLoading] = useState(false);
+  const [err, setErr] = useState<string | undefined>(undefined);
+  const refetchInterval = 10000;
 
-const NodeDetails: NextPage<NodeDetailsData> = ({ viewModel, err }) => {
   const { TabPane } = Tabs;
   const { Option } = Select;
+  const router = useRouter();
   const { query } = useRouter();
 
   // neither of these values will ever be null because the URL path depends on them to render this page
-  const { gatewayUID, nodeId } = query as SparrowQueryInterface;
+  const { gatewayUID, nodeId, minutesBeforeNow } =
+    query as SparrowQueryInterface;
   const nodeUrl = `/${gatewayUID}/node/${nodeId}/details`;
 
-  const router = useRouter();
-  // Call this function whenever you want to
-  // refresh props!
-  const refreshData = async () => {
-    await router.replace(router.asPath);
-  };
+  const {
+    isLoading: gatewayLoading,
+    error: gatewayError,
+    data: gateway,
+  } = useGateway(gatewayUID);
+
+  const {
+    isRefetching: nodeRefetching, // for when we update the node name
+    isLoading: nodeLoading,
+    error: nodeError,
+    data: node,
+    refetch: nodeRefetch,
+  } = useNode(gatewayUID, nodeId, refetchInterval);
+
+  const {
+    isLoading: readingsLoading,
+    error: readingsError,
+    data: readings,
+    refetch: nodeReadingsRefetch,
+  } = useNodeData(gatewayUID, nodeId, minutesBeforeNow, refetchInterval);
+
+  useEffect(() => {
+    if (gateway && node && readings) {
+      const nodeModel: NodeDetailViewModel = getNodeDetailsPresentation(
+        node,
+        gateway,
+        readings
+      );
+      setViewModel(nodeModel);
+    }
+  }, [gateway, node, readings]);
+
+  useEffect(() => {
+    if (gatewayLoading) {
+      setIsLoading(true);
+    }
+    if (nodeLoading || nodeRefetching) {
+      setIsLoading(true);
+    }
+    if (readingsLoading) {
+      setIsLoading(true);
+    }
+    if (!gatewayLoading && !nodeLoading && !readingsLoading) {
+      setIsLoading(false);
+    }
+  }, [gatewayLoading, nodeLoading, nodeRefetching, readingsLoading]);
```

Comment:
I get uneasy around this many `useEffect`s because working on dev.blues.io we ended up using a lot more useEffect than we needed. Every time we thought 'oh when this changes we need that to change' we added another useEffect when sometimes a `const` would do just fine.  Feels like we're making an awfully complicated program out of the react `useEffect` system. Maybe if we could pull the error handling somehow... I need to sleep on it.